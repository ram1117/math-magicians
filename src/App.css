.App {
  text-align: center;
}

.nav-bar {
  display: flex;
  padding: 0 2%;
  background: #353535;
  align-items: center;
}

.app-logo {
  width: 60%;
  color: #16a085;
}

.nav-items > *,
footer > p {
  color: #16a085;
  font-weight: 700;
  text-decoration: none;
}

.nav-items {
  padding: 5% 0;
}

.nav-items :hover {
  color: #fff;
}

footer {
  width: 100%;
  position: fixed;
  bottom: 0;
  text-align: center;
  background: #353535;
}

.profile-link {
  color: #fff;
}

.nav-list {
  width: 40%;
  list-style: none;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

@media screen and (min-width: 768px) {
  .nav-list {
    flex-direction: row;
    list-style: none;
    display: flex;
    justify-content: space-around;
  }
}
