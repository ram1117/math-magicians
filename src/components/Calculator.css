.calc-container {
  display: grid;
  height: 80%;
  width: 90%;
  margin: 0 auto;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: repeat(6, 1fr);
  color: #16a085;
}

.calc-display {
  height: 1.6em;
  grid-column: 1 /span 4;
  border: solid 2px #c0c0c0;
  padding: 3% 5%;
  text-align: right;
  background: #353535;
  font-size: 1.4em;
}

.buttons {
  background: #353535;
  border: solid 2px #c0c0c0;
  font-size: 1.4em;
  color: #16a085;
}

.buttons:hover {
  cursor: pointer;
}

.buttons:nth-child(5),
.buttons:nth-child(9),
.buttons:nth-child(13),
.buttons:nth-child(17),
.buttons:nth-child(20) {
  background: #f5913e;
}

.buttons:nth-child(18) {
  grid-column: 1/ span 2;
}
