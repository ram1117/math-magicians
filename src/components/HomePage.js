const HomePage = () => (
  <>
    <h2>Welcome to Math Magician Page!!!</h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industry&apos;s standard dummy text ever since the 1500s,
      when an unknown printer took a galley of type and scrambled it to make a type specimen book.
      It has survived not only five centuries,
      but also the leap into electronic typesetting, remaining essentially unchanged.
      and more recently with desktop publishing software
      like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
    <p>
      It is a long established fact that a reader will be distracted by the readable content of a
      page when looking at its layout. The point of using Lorem Ipsum is that it has a
      more-or-less normal distribution of letters,
      as opposed to using &lsquo;Content here, content here&rsquo;,
      making it look like readable English.
      Many desktop publishing packages and web page editors now
      use Lorem Ipsum as their default model text,
      and a search for &lsquo;lorem ipsum&rsquo; will uncover many web sites still in their infancy.
      Various versions have evolved over the years, sometimes by accident,
      sometimes on purpose (injected humour and the like).
    </p>
  </>
);

export default HomePage;
