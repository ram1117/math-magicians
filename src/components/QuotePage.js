import styles from '../styles/QuotePage.module.css';

const QuotePage = () => (
  <>
    <blockquote className={styles['quote-block']}>
      “Mathematics as an expression of the human mind reflects the active will,
      the contemplative reason, and the desire for aesthetic perfection.
      Its basic elements are logic and intuition, analysis and construction,
      generality and individuality.”
      <br />
      <span data-testid="quote-author">— Richard Courant</span>
    </blockquote>
  </>
);

export default QuotePage;
