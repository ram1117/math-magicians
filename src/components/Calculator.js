import { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calcData, setCalcData] = useState({ total: '0', next: '', operation: '' });
  const handleClick = (event) => {
    setCalcData(calculate(calcData, event.target.textContent));
  };
  const calcBtnText = ['AC', '+/-', '%', 'รท', '7', '8', '9',
    'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div className="calc-container">
      <>
        <div className="calc-display" data-testid="calc-display">
          {calcData.total}
          {calcData.operation}
          {calcData.next}
        </div>
      </>
      {
        calcBtnText.map((item) => <button className="buttons" onClick={(event) => { handleClick(event); }} key={item} type="button">{item}</button>)
      }
    </div>
  );
};
export default Calculator;
