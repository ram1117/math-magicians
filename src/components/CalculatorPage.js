import styles from '../styles/CalculatorPage.module.css';
import Calculator from './Calculator';

const CalculatorPage = () => (
  <>
    <div className={styles.wrapper}>
      <h2>Let&apos;s do some Math Magic</h2>
      <Calculator className={styles.calculator} />
    </div>
  </>
);

export default CalculatorPage;
