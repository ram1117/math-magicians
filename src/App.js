import './App.css';
import { Link, Routes, Route } from 'react-router-dom';
import HomePage from './components/HomePage';
import CalculatorPage from './components/CalculatorPage';
import QuotePage from './components/QuotePage';

const App = () => {
  const ghProfile = 'https://github.com/ram1117?tab=repositories';
  return (
    <>
      <nav className="nav-bar">
        <h2 className="app-logo">Math Magician</h2>
        <ul className="nav-list">
          <li className="nav-items"><Link to="/">Home</Link></li>
          <li className="nav-items"><Link to="calculator">Calculator</Link></li>
          <li className="nav-items"><Link to="quotes">Quote</Link></li>
        </ul>
      </nav>
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="calculator" element={<CalculatorPage />} />
        <Route path="quotes" element={<QuotePage />} />
      </Routes>
      <footer>
        <p>
          Developed and maintained by:
          <Link className="profile-link" to={ghProfile} target="_blank"> @ram1117</Link>
        </p>
      </footer>
    </>
  );
};

export default App;
