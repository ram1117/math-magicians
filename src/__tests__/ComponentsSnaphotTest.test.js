import TestRenderer from 'react-test-renderer';
import App from '../App.js';
import Calculator from '../components/Calculator.js';
import HomePage from '../components/HomePage.js';
import QuotePage from '../components/QuotePage.js';
import CalculatorPage from '../components/CalculatorPage';
import { BrowserRouter } from 'react-router-dom';


describe('Snapshot testing components:',()=>{
  it('App renders correctly',()=>{
    const app = TestRenderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(app).toMatchSnapshot();
  });

  it('Home Page renders correctly',()=>{
    const homePage = TestRenderer.create(<HomePage />).toJSON();
    expect(homePage).toMatchSnapshot();
  });

  it('Calculator Page renders correctly',()=>{
    const calcPage = TestRenderer.create(<CalculatorPage />).toJSON();
    expect(calcPage).toMatchSnapshot();
  });

  it('HomePage renders correctly',()=>{
    const quotePage = TestRenderer.create(<QuotePage />).toJSON();
    expect(quotePage).toMatchSnapshot();
  });
  it('calculator renders correctly', () => {
    const calc = TestRenderer
      .create(<Calculator />)
      .toJSON();
  
    expect(calc).toMatchSnapshot();
  });
});
