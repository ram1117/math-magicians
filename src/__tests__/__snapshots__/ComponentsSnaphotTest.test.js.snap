// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing components: App renders correctly 1`] = `
Array [
  <nav
    className="nav-bar"
  >
    <h2
      className="app-logo"
    >
      Math Magician
    </h2>
    <ul
      className="nav-list"
    >
      <li
        className="nav-items"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-items"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="nav-items"
      >
        <a
          href="/quotes"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="wrapper"
  >
    <h2
      className="pageHeader"
      data-testid="home-header"
    >
      Welcome to Math Magician Page!!!
    </h2>
    <p>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
    <p>
      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using ‘Content here, content here’, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for ‘lorem ipsum’ will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
    </p>
  </div>,
  <footer>
    <p>
      Developed and maintained by:
      <a
        className="profile-link"
        href="https://github.com/ram1117?tab=repositories"
        target="_blank"
      >
         @ram1117
      </a>
    </p>
  </footer>,
]
`;

exports[`Snapshot testing components: Calculator Page renders correctly 1`] = `
<div
  className="wrapper"
>
  <h2
    data-testid="calc-header"
  >
    Let's do some Math Magic
  </h2>
  <div
    className="calc-container"
  >
    <div
      className="calc-display"
      data-testid="calc-display"
    >
      0
    </div>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="buttons"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Snapshot testing components: Home Page renders correctly 1`] = `
<div
  className="wrapper"
>
  <h2
    className="pageHeader"
    data-testid="home-header"
  >
    Welcome to Math Magician Page!!!
  </h2>
  <p>
    Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
  </p>
  <p>
    It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using ‘Content here, content here’, making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for ‘lorem ipsum’ will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
  </p>
</div>
`;

exports[`Snapshot testing components: HomePage renders correctly 1`] = `
<blockquote
  className="quote-block"
>
  “Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.”
  <br />
  <span
    data-testid="quote-author"
  >
    — Richard Courant
  </span>
</blockquote>
`;

exports[`Snapshot testing components: calculator renders correctly 1`] = `
<div
  className="calc-container"
>
  <div
    className="calc-display"
    data-testid="calc-display"
  >
    0
  </div>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    ÷
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="buttons"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
